pkg/util/sysutil/sysutil_unix.go

```diff
@@ -55,3 +57,18 @@ func StatFS(path string) (*FSInfo, error) {
 		BlockSize:   int64(fs.Bsize),
 	}, nil
 }
+
+// StatAndLinkCount wraps os.Stat, returning its result and, if the platform
+// supports it, the link-count from the returned file info.
+func StatAndLinkCount(path string) (os.FileInfo, int64, error) {
+	stat, err := os.Stat(path)
```

Comment:
Given that you’re calling os.Stat in both implementations, I would have left that in the caller and added a LinkCount function that takes an os.FileInfo. Not a big deal. This is fine as well.

Reply:
I didn't really did into it to far, but at first glance it looked like if/when we want to get the link count on windows, we'd need to do our own syscall, since it didn't look like it is in the windows version of `FileInfo.Sys()`'s returned value: https://github.com/golang/go/blob/master/src/os/types_windows.go#L142

I think it _is_ in a `ByHandleFileInformation` which I think we'd need the path to get.